overlord/aspectstate/aspectstate.go

```diff
@@ -22,83 +22,125 @@ import (
 	"errors"
 
 	"github.com/snapcore/snapd/aspects"
-	"github.com/snapcore/snapd/overlord/aspectstate/aspecttest"
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/overlord/assertstate"
 	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/strutil"
 )
 
 // SetAspect finds the aspect identified by the account, bundleName and aspect
-// and sets the specified field to the supplied value in the provided matching databag.
-func SetAspect(databag aspects.DataBag, account, bundleName, aspect, field string, value interface{}) error {
-	accPatterns := aspecttest.MockWifiSetupAspect()
-	schema := aspects.NewJSONSchema()
-
-	aspectBundle, err := aspects.NewBundle(account, bundleName, accPatterns, schema)
+// and sets the request fields to their respective values.
+func SetAspect(st *state.State, account, bundleName, aspect string, requests map[string]interface{}) error {
+	bundle, err := getBundle(st, account, bundleName)
 	if err != nil {
 		return err
 	}
 
-	asp := aspectBundle.Aspect(aspect)
+	asp := bundle.Aspect(aspect)
 	if asp == nil {
+		keys := make([]string, 0, len(requests))
+		for k := range requests {
+			keys = append(keys, k)
+		}
+
 		return &aspects.NotFoundError{
 			Account:    account,
 			BundleName: bundleName,
 			Aspect:     aspect,
 			Operation:  "set",
-			Request:    field,
+			Request:    strutil.Quoted(keys),
```

Comment:
ah, it would make more sense perhaps to make Request into Requests []string ?

Reply:
ah sorry, I didn't see this comment before replying to the other one. Yes, I'll change the field to reflect the new usage as well